### PR TITLE
Fix build errors with the ORIGINAL_RATINGS flag

### PR DESCRIPTION
**Pull request by Big-Blue-1**
Fixing some build errors when using the ORIGINAL_RATINGS flag, since nobody applied the same changes that the rest of the code got.

### CODE REVIEW

**Review by Gymnasiast — CHANGES_REQUESTED** [review]
Code looks fine, please add a changelog entry:

`Fix: [#21937] Build errors with the ORIGINAL_RATINGS flag.`

**Review by Gymnasiast — APPROVED** [review]

File: src/openrct2/ride/RideRatings.cpp [hunk 1]
```diff
@@ -1047,12 +1047,12 @@ static void RideRatingsCalculate(RideRatingUpdateState& state, Ride& ride)
...
-    if (ride.ratings.excitement != -1)
+    if (ride.ratings.Excitement != -1)
     {
         // Address underflows allowed by original RCT2 code
-        ride.ratings.excitement = max(0, ride.ratings.excitement);
-        ride.ratings.intensity = max(0, ride.ratings.intensity);
-        ride.ratings.nausea = max(0, ride.ratings.nausea);
+        ride.ratings.Excitement = std::max(0, (int)ride.ratings.Excitement);
+        ride.ratings.Intensity = std::max(0, (int)ride.ratings.Intensity);
+        ride.ratings.Nausea = std::max(0, (int)ride.ratings.Nausea);
```
Comment (tupaschoal) [inline src/openrct2/ride/RideRatings.cpp:1055]: Why was the cast to `int` needed?

If it is really needed, please don't use a c-style cast, but rather a named cast:
```suggestion
        ride.ratings.Excitement = std::max(0, static_cast<int>(ride.ratings.Excitement));
        ride.ratings.Intensity = std::max(0, static_cast<int>(ride.ratings.Intensity));
        ride.ratings.Nausea = std::max(0, static_cast<int>(ride.ratings.Nausea));
```

Comment (Gymnasiast) [reply src/openrct2/ride/RideRatings.cpp:1055]: uint16_t would be better. That is what the ratings actually use. Also we can use templating on std::max.

```suggestion
        ride.ratings.Excitement = std::max<uint16_t>(0, ride.ratings.Excitement);
        ride.ratings.Intensity = std::max<uint16_t>(0, ride.ratings.Intensity);
        ride.ratings.Nausea = std::max<uint16_t>(0, ride.ratings.Nausea);
```

Comment (Big-Blue-1) [reply src/openrct2/ride/RideRatings.cpp:1055]: yeah i just noticed that too and tried to look into the datatype of ratings. i will change it to <uint16_t>.

**Review by Gymnasiast — APPROVED** [review]

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**